wa/framework/workload.py

```diff
@@ -351,49 +354,76 @@ def __init__(self, owner, install_timeout=300, version=None, variant=None,
         self.apk_version = None
         self.logcat_log = None
 
-    def init_resources(self, resolver):
-        self.apk_file = resolver.get(ApkFile(self.owner,
-                                             variant=self.variant,
-                                             version=self.version),
-                                             strict=self.strict)
-        self.apk_info = ApkInfo(self.apk_file)
-
     def setup(self, context):
+        self.resolve_package(context)
```

Comment:
This should b edone as part of `initialize` rather than `setup` as this only needs to happen once per instance per run.